### ISSUE TITLE
Ignore wayland scaling error

### ISSUE THREAD

**Issue by u16rogue** [Genymobile/scrcpy#4686]
- [x] I have checked that a similar [feature request](https://github.com/Genymobile/scrcpy/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22) does not already exist.
* with keywords: `is:open is:issue label:"feature request" wayland, scale`

**Is your feature request related to a problem? Please describe.**
- My phone no longer has a functioning LCD screen so scrcpy has been a great help. but sometimes when I have to do some things outside my computer I leave it locked and toggle my display off. Whenever I come back scrcpy closes itself due to:
```
wl_surface@32: error 0: Specified scale value (0) is not positive
WARN: Killing the server...
```

**Describe the solution you'd like**
- For scrcpy to ignore this error.

**Describe alternatives you've considered**
- n/a

**Additional context**
Log:
```
scrcpy 2.3.1 <https://github.com/Genymobile/scrcpy>
INFO: ADB device found:
INFO:     -->   (usb)  <>              device  <>
INFO:         (tcpip)  <>              device  <>
/usr/share/scrcpy/scrcpy-server: 1 file pushed, 0 skipped. 379.9 MB/s (66007 bytes in 0.000s)
adb: error: more than one device/emulator
ERROR: "adb reverse" returned with value 1
WARN: 'adb reverse' failed, fallback to 'adb forward'
[server] INFO: Device: [samsung] samsung <> (Android 7.0)
[server] WARN: Audio disabled: it is not supported before Android 11
INFO: Renderer: opengl
INFO: OpenGL version: 4.6 (Compatibility Profile) Mesa 23.3.5-arch1.1
INFO: Trilinear filtering enabled
WARN: Demuxer 'audio': stream explicitly disabled by the device
INFO: Texture: 1080x1920
wl_surface@32: error 0: Specified scale value (0) is not positive
WARN: Killing the server...
```

**Comment by rom1v** [comment]
Please report to SDL, with more details about your desktop environment.

https://github.com/libsdl-org/SDL

**Comment by u16rogue** [comment]
This issue has been resolved on SDL's side (https://github.com/libsdl-org/SDL/issues/9107). Should be available (soon) on SDL 2.32.
